testsuite/MDAnalysisTests/core/test_universe.py

```diff
@@ -707,6 +707,63 @@ def test_add_connection(self, universe, attr, values):
     def add_connection_error(self, universe, attr, values):
         with pytest.raises(ValueError):
             universe.add_TopologyAttr(attr, values)
+
+
+class TestDelTopologyAttr(object):
+    @pytest.fixture()
+    def universe(self):
+        u = make_Universe(("masses", "charges", "resnames"))
+        u.add_TopologyAttr("bonds", [(0, 1)])
+        return u
+
+    def test_del_TA_fail(self, universe):
+        with pytest.raises(ValueError, match="Unrecognised"):
+            universe.del_TopologyAttr('silly')
+
+    def test_absent_fail(self, universe):
+        with pytest.raises(ValueError, match="not in Universe"):
+            universe.del_TopologyAttr("angles")
+
+    def test_wrongtype_fail(self, universe):
+        with pytest.raises(ValueError, match="must be str or TopologyAttr"):
+            universe.del_TopologyAttr(list)
+
+    @pytest.mark.parametrize(
+        'todel,attrname', [
+            ("charge", "charges"),
+            ("charges", "charges"),
+            ("bonds", "bonds"),
+        ]
+    )
+    def test_del_str(self, universe, todel, attrname):
+        assert hasattr(universe.atoms, attrname)
+        universe.del_TopologyAttr(todel)
+        assert not hasattr(universe.atoms, attrname)
+
+    def test_del_attr(self, universe):
+        assert hasattr(universe.atoms, "resnames")
```

Comment:
One thing I'd want to check is if I have an AtomGroup object persisting around the `del_TA` call will the attribute still fail correctly?

i.e.
```python
ag = u.atoms[]
ag.resnames = 'xyz'
u.del_TopologyAttr('resnames')
ag.resnames <- is this still failing correctly?
```

I'm stlil a little unsure about how our monkeypatching works at times.

Reply:
Now covered in `test_del_attr_from_ag`